firedrake/adjoint/blocks.py

```diff
@@ -245,9 +250,12 @@ def evaluate_hessian_component(self, inputs, hessian_inputs, adj_inputs,
         return hessian_inputs[0]
 
     def recompute(self):
-        dep = self.get_dependencies()[0].checkpoint
-        output = self.get_outputs()[0].checkpoint
-        self.backend.Function.assign(self.backend.Function.sub(output, self.idx), dep)
+        deps = self.get_dependencies()
+        sub_func = deps[0].checkpoint
```

Comment:
Is there a reason for using `checkpoint` instead of `saved_output` as it is usually done in `recompute` ?